src/expire.c

```diff
@@ -503,9 +503,16 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
 
     if (getLongLongFromObjectOrReply(c, param, &when, NULL) != C_OK)
         return;
-
+    if (when <= 0) {
+        addReplyErrorFormat(c, "invalid expire time in %s", c->cmd->name);
+        return;
+    }
```

Comment:
@oranagra 

This caches negative expiry before doing `    if (unit == UNIT_SECONDS) when *= 1000;`

Before:
```
127.0.0.1:6379> SET A B
OK
127.0.0.1:6379> EXPIRE A -9999999999999999
(integer) 1
127.0.0.1:6379> GET A
"B"
127.0.0.1:6379> TTL A
(integer) 8446744073709547
```

After:
```
127.0.0.1:6379> SET A B
OK
127.0.0.1:6379> EXPIRE A -9999999999999999
(error) ERR invalid expire time in expire
127.0.0.1:6379> GET A
"B"
```